File: src/main/java/com/empiricus/statusviajante/repository/ViagemRepository.java
package com.empiricus.statusviajante.repository;

import com.empiricus.statusviajante.model.GastoViagemModel;
import com.empiricus.statusviajante.model.ViagemModel;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface ViagemRepository extends JpaRepository<ViagemModel, Long> {

    public Optional<ViagemModel> findById(Long idViagem);

    public List<ViagemModel> findByUsuario_idUsuario(Long idViagem);
}

File: src/main/java/com/empiricus/statusviajante/repository/ViagemRepository.kt
package com.empiricus.statusviajante.repository

import com.empiricus.statusviajante.model.ViagemModel
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Transactional
import java.util.*

@Repository
interface ViagemRepository : JpaRepository<ViagemModel?, Long?> {

    fun findById(idViagem: Long): Optional<ViagemModel?>

    fun findByUsuario_idUsuario(idViagem: Long?): List<ViagemModel?>?

    fun findByUsuario_idUsuarioAndIdViagem(idUsuario: Long?, idViagem: Long): Optional<ViagemModel?>

    @Transactional
    fun deleteByUsuario_idUsuario(idUsuario: Long?) : Int

}